src/app/asset-page/asset-page.component.pug

```diff
@@ -118,43 +117,19 @@
-              h6.dropdown-header(*ngIf="assets[0].disableDownload") Item not available for download
-              a#downloadAssetLink.dropdown-item(download,
-                [attr.href]="userHasAcceptedTerms() ? generatedFullURL : null",
-                (click)="handleDownloadImageClick()",
-                (keydown.enter)="handleDownloadImageClick()",
-                (keyup.arrowdown)="downloadOptsArrowDown($event.target)",
-                (keyup.arrowup)="downloadOptsArrowUp($event.target)",
-                [class.disabled]="assets[0].disableDownload",
-                title="{{ assets[0].disableDownload ? 'Image is not available for download.' : 'Image download'}}",
-                target="_blank",
-                tabindex="2",
-                aria-label="Download item",
-                role="button")
-                i.icon.icon-download
-                | &nbsp;Download item
-              a#downloadViewLink.dropdown-item(
-                [attr.href]="userHasAcceptedTerms() ? downloadViewLink : null",
-                (click)="handleDownloadViewClick()",
-                (keydown.enter)="handleDownloadViewClick()",
-                (keyup.arrowup)="downloadOptsArrowUp($event.target)",
-                [class.disabled]="!downloadViewReady || assets[0].disableDownload || assets[0].typeName !== 'image'",
-                target="_blank",
-                tabindex="2",
-                aria-label="Download item’s detail view",
-                role="button")
-                i.icon.icon-download-detail-view
-                | &nbsp;Download detail view
-          //- False door button if user is not logged in
-          button.assetpage-btn.btn.btn-secondary(*ngIf="assets[0].downloadLink && !(user.isLoggedIn || hasExternalAccess) && !assets[0].publicDownload", (click)="showLoginModal = true", (keydown.enter)="showLoginModal = true", tabindex="2", aria-label="Download")
-            | {{ 'ASSET_PAGE.BUTTONS.DOWNLOAD' | translate}}
-            i#downloadAssetLink.icon.icon-download-asset-orange
+          ng-container(*ngIf="requireLoginForAdlDownload")
+            ng-container(*ngIf="user.isLoggedIn || assets[0].publicDownload")
+              ng-container(*ngTemplateOutlet="downloadDropdown")
+            ng-container(*ngIf="!user.isLoggedIn && !assets[0].publicDownload")
+              ng-container(*ngTemplateOutlet="falseDoorDownload")
+          ng-container(*ngIf="!requireLoginForAdlDownload")
+            ng-container(*ngIf="assets[0].publicDownload")
+              ng-container(*ngTemplateOutlet="downloadDropdown")
+            ng-container(*ngIf="!assets[0].publicDownload")
+              ng-container(*ngIf="(user?.status || user.isLoggedIn)")
+                ng-container(*ngTemplateOutlet="downloadDropdown")
+              ng-container(*ngIf="!(user?.status || user.isLoggedIn)")
+                ng-container(*ngTemplateOutlet="falseDoorDownload")
```

Comment:
There are quite a few branches to the `if` condition here. Wondering if we could consolidate and simplify this a bit. It seems like the template really has two things it cares about 1) should I show the `downloadDropdown` or 2) the `falseDoorDownload` button? Maybe we can move some of the boolean logic to a function that can tell us `canUserDownload` (or something like that) and use it here in the template? That can take into consideration things like the `publicDownload`, `downloadLink`, `requireLoginForAdlDownload`, `isLoggedIn`, etc. etc.

I also feel like there is an opportunity to consolidate some of the boolean logic, but I am not exactly sure how that would look right now. I help take a look too if you want!

Reply:
I pushed a much better solution in b5f3fcd 👍